great_expectations_cloud/agent/agent.py

```diff
@@ -144,43 +229,57 @@ def run(self) -> None:
         """Open a connection to GX Cloud."""
 
         print("Opening connection to GX Cloud.")
-        self._listen()
+        asyncio.run(self._listen())
         print("The connection to GX Cloud has been closed.")
 
     # ZEL-505: A race condition can occur if two or more agents are started at the same time
     #          due to the generation of passwords for rabbitMQ queues. This can be mitigated
     #          by adding a delay and retrying the connection. Retrying with new credentials
     #          requires calling get_config again, which handles the password generation.
+    # Note: This is not the number of retries of processing a message, it's the number of retries
+    #       to establish a connection to the broker.
     @retry(
         retry=retry_if_exception_type((AuthenticationError, ProbableAuthenticationError)),
         wait=wait_exponential(multiplier=1, min=1, max=10),
         stop=stop_after_attempt(3),
         after=after_log(LOGGER, logging.DEBUG),
     )
-    def _listen(self) -> None:
+    async def _listen(self) -> None:
         """Manage connection lifecycle."""
-        subscriber = None
         try:
-            client = AsyncRabbitMQClient(url=str(self._config.connection_string))
-            subscriber = Subscriber(client=client)
-            print("The GX Agent is ready.")
-            # Open a connection until encountering a shutdown event
-            subscriber.consume(
-                queue=self._config.queue,
-                on_message=self._handle_event_as_thread_enter,
+            broker = RabbitBroker(
+                url=str(self._config.connection_string),
             )
+            app = FastStream(broker)
+            queue = RabbitQueue(name=self._config.queue, durable=True, passive=True)
+            print("Queue is valid.")
+
+            # FastStream declares default exchange if not provided
+            @broker.subscriber(queue, retry=MAX_DELIVERY)
+            async def handle_me(
+                msg: dict, correlation_id: str = Context("message.correlation_id")
+            ) -> None:
+                print(f"Received: {msg}")
+                print(f"Correlation ID: {correlation_id}")
+                await handler(msg, gx_agent=self, correlation_id=correlation_id)
+
+            print("FastStream is ready.")
+            await app.run()
         except KeyboardInterrupt:
             print("Received request to shut down.")
-        except (SubscriberError, ClientError):
+        except (
+            SubscriberError,
+            ClientError,
+            asyncio.exceptions.CancelledError,
+            aiormq.exceptions.ChannelAccessRefused,
+        ):
             print("The connection to GX Cloud has encountered an error.")
+            print(traceback.format_exc())
```

Comment:
Is there any reason to not surface this error in logs? I can't think of one, just checking.

Reply:
Pushing fix

Reply:
addressed in 5fcd42b